lib/mergeset/part_header.go

```diff
@@ -136,7 +136,6 @@ func (ph *partHeader) ParseFromPath(partPath string) error {
 	if ph.itemsCount != phj.ItemsCount {
 		return fmt.Errorf("invalid ItemsCount in %q; got %d; want %d", metadataPath, phj.ItemsCount, ph.itemsCount)
 	}
-	ph.blocksCount = phj.BlocksCount
```

Comment:
No, this assignment is a pure typo, which has been introduced since the inception of VictoriaMetrics. The original assignment to `ph.blocksCount` happens at line 116 above. This assignment should be removed, since it suppresses the check at line 136.